### PR TITLE
add better health endpoints

### PR DESCRIPTION
**Pull request by justincorrigible**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "typescript-transform-paths": "^3.3.1"
   },
   "dependencies": {
-    "@caravinci/arranger-server": "^2.15.4",
+    "@caravinci/arranger-server": "^2.15.5",
```
Comment (justincorrigible) [inline package.json:37]: Arranger's 'ping' path can now be customised. Responds with a better message than 'ok'

File: src/constants/endpoint.ts [hunk 2]
```diff
@@ -1,3 +1,2 @@
 export const BASE_ENDPOINT = '/api';
-
-export const ARRANGER_READY_ENDPOINT = '/ping';
+export const HEALTH_ENDPOINT = '/health';
```
Comment (justincorrigible) [inline src/constants/endpoint.ts:2]: rename the constant, and make it useful for both health endpoints

File: src/index.ts [hunk 3]
```diff
@@ -5,12 +5,12 @@ require('dotenv').config({
...
 import { Router } from 'express';
 
-import { BASE_ENDPOINT } from './constants/endpoint';
+import { BASE_ENDPOINT, HEALTH_ENDPOINT } from './constants/endpoint';
 import app from './server';
 
 const port = Number(process.env.PORT || 4000);
 
-Arranger().then((arrangerRouter: Router) => {
+Arranger({ pingPath: HEALTH_ENDPOINT }).then((arrangerRouter: Router) => {
```
Comment (justincorrigible) [inline src/index.ts:13]: customise the 'ping' endpoint to `/api/health`

File: src/routes/health/health.router.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ const router: Router = Router();
 
 router.get('/', (req, res) => {
   res.status(200).send({
-    status: 'API Server is running...',
+    message: 'API Server is running...',
```
Comment (justincorrigible) [inline src/routes/health/health.router.ts:7]: status is 200, message is whatever else.

File: src/routes/index.ts [hunk 5]
```diff
@@ -1,16 +1,17 @@
 import { Router } from 'express';
 
-import AuthRouter from './auth/auth.router';
-import HealthRouter from './health/health.router';
-import StorageRouter from './storage/storage.router';
+import authRouter from './auth/auth.router';
+import storageRouter from './storage/storage.router';
 
-const Routes = (): Router => {
+const routes = (): Router => {
```
Comment (justincorrigible) [inline src/routes/index.ts:6]: none of these are classes. feel free to disregard, but we should keep consistent naming

File: src/routes/index.ts [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
   const router = Router();
 
-  router.use('/auth', AuthRouter);
-  router.use('/health', HealthRouter);
```
Comment (justincorrigible) [inline src/routes/index.ts:11]: un-nested server `/health` from `/api`

File: src/server.ts [hunk 7]
```diff
@@ -26,7 +26,16 @@ app.use(
...
 
 // Handle logs in console during development
-app.use(morgan('dev'));
+app.use(
+  morgan('dev', {
+    skip: (req, res) => {
+      // logs everything but health checks on dev, errors only otherwise
+      return process.env.NODE_ENV === 'development' || process.env.DEBUG === 'true'
+        ? req.originalUrl.includes(HEALTH_ENDPOINT)
+        : res.statusCode < 400;
```
Comment (justincorrigible) [inline src/server.ts:35]: restored the specified logging, as it's going to get pretty loud while debugging in cluster.

File: src/server.ts [hunk 8]
```diff
@@ -37,8 +46,9 @@ if (process.env.NODE_ENV === 'production') {
 /************************************************************************************
  *                               Register all routes
  ***********************************************************************************/
-const routes = Routes();
-app.use(BASE_ENDPOINT, routes);
+
+app.use(BASE_ENDPOINT, apiRoutes());
+app.use(HEALTH_ENDPOINT, healthRouter);
```
Comment (justincorrigible) [inline src/server.ts:51]: added the server health endpoint at root.

File: src/routes/index.ts [hunk 9]
```diff
@@ -1,16 +1,17 @@
...
+  router.use('/auth', authRouter);
+  router.use('/storage', storageRouter);
 
   return router;
 };
-export default Routes;
+
+export default routes;
+
+export { default as healthRouter } from './health/health.router';
```
Comment (joneubank) [inline src/routes/index.ts:17]: 👍

**Review by joneubank — APPROVED** [review]

**Pull request merged**